src/finalizing/derive-latest.js

```diff
@@ -0,0 +1,76 @@
+import fs, { readdirSync } from 'fs';
+import sh from 'shelljs';
+import { globSync } from 'glob';
+import semver from 'semver';
+
+/**
+ * @description Among the list of versions, gets the latest version and adds it to the latest directory in dist/do-type/do-name/latest
+ * @param {*} context
+ * @return returns the latest version
+ */
+
+export async function deriveLatest(context) {
+  //Extracting path
+  const glob = '**/graph.ttl';
+  const digitalObjects = globSync(glob, { cwd: context.deployHome }).map((p) => p.split('/').slice(0, -2).join('/'));
+
+  for (const digitalObject of digitalObjects) {
+    //Getting list of versions
+    const versions = readdirSync(digitalObject).filter((v) => v != 'latest');
+
+    const obj = digitalObject.split('/').slice(0, 3).join('/');
+    try {
+      if (fs.existsSync(`${obj}/latest`)) {
+        sh.rm('-r', `${obj}/latest`);
+      }
+    } catch (error) {
+      console.error('Error while deleting latest directory', error);
+    }
+    let sortedVersions = versions;
+
+    if (versions.length > 1) {
+      //Sorting the version in descending order
+      sortedVersions = versions.sort(async (a, b) => {
+        let versionA = a.replace('v', '');
+        let versionB = b.replace('v', '');
+
+        versionA = await validateVersion(versionA);
+        versionB = await validateVersion(versionB);
+
+        // Skip processing invalid versions
+        if (!semver.valid(versionA) || !semver.valid(versionB)) {
```

Comment:
It's not an error when this happens. Again, just filter out the invalid directories.